client/examples/repl/command.rs

```diff
@@ -288,7 +287,7 @@ impl Command for Slip10DeriveCommand {
 
         client.execute_procedure(Slip10Derive {
             curve: Curve::Ed25519,
-            chain: Chain::from_u32_hardened(chain_code.parse()),
+            chain: vec![chain_code.parse().map(|s: u32| s.harden().into()).unwrap()],
```

Comment:
Why unwrap here?

Reply:
Otherwise it can't deduce the result type that implements Segment.

Reply:
That's not really relevant. Previously we had an infallable process and now we unwrap.

Reply:
Previously parse error was silenced (unless it was an expected behaviour). Added parse error handler.